libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessageParser.java

```diff
@@ -126,7 +126,7 @@ public boolean parse(ByteBuffer buffer) throws IOException {
                     continue;
 
                 // Garbled message
-                if (FIXCheckSums.sum(buffer, beginning, position - beginning + length) % 256
+                if ((FIXCheckSums.sum(buffer, beginning, position - beginning + length) & (256 - 1))
```

Comment:
Perhaps we could use `& 0xff` here, as the hexadecimal literals pair well with bitwise operations?

Reply:
Good idea! 💥 Was too focussed on keeping the existing `256` visible with first `255` then `256 - 1`.